spiketoolkit/sorters/klusta/klusta.py

```diff
@@ -38,127 +28,113 @@ def klusta(
-
-    print('Elapsed time: ', time.time() - t_start_proc)
-
-    return sorting
-
-
-def _klusta(
-        recording,  # The recording extractor
-        output_folder=None,
-        probe_file=None,
-        file_name=None,
-        adjacency_radius=None,
-        threshold_strong_std_factor=5,
-        threshold_weak_std_factor=2,
-        detect_sign=-1,
-        extract_s_before=16,
-        extract_s_after=32,
-        n_features_per_channel=3,
-        pca_n_waveforms_max=10000,
-        num_starting_clusters=50
-):
-    try:
-        import klusta
-        import klustakwik2
-    except ModuleNotFoundError:
-        raise ModuleNotFoundError("\nTo use Klusta, install klusta and klustakwik2: \n\n"
-                                  "\npip install klusta klustakwik\n"
-                                  "\nMore information on klusta at: "
-                                  "\nhttps://github.com/kwikteam/phy"
-                                  "\nhttps://github.com/kwikteam/klusta")
-    source_dir = Path(__file__).parent
-    if output_folder is None:
-        output_folder = Path('klusta')
-    else:
-        output_folder = Path(output_folder)
-    if not output_folder.is_dir():
-        output_folder.mkdir()
-
-    # save prb file:
-    if probe_file is None:
-        probe_file = output_folder / 'probe.prb'
-        se.saveProbeFile(recording, probe_file, format='klusta', radius=adjacency_radius)
-
-    # save binary file
-    if file_name is None:
-        file_name = Path('recording')
-    elif file_name.suffix == '.dat':
-        file_name = file_name.stem
-    se.writeBinaryDatFormat(recording, output_folder / file_name)
-
-    if detect_sign < 0:
+    """
+    
+    sortername = 'klusta'
+    installed = HAVE_KLUSTA
+    SortingExtractor_Class = se.KlustaSortingExtractor
+    
+    _default_params = {
+        'file_name': None,
+        'probe_file': None,
+    
+        'adjacency_radius': None,
+        'threshold_strong_std_factor': 5,
+        'threshold_weak_std_factor': 2,
+        'detect_sign': -1,
+        'extract_s_before': 16,
+        'extract_s_after': 32,
+        'n_features_per_channel': 3,
+        'pca_n_waveforms_max': 10000,
+        'num_starting_clusters': 50,
+        'parallel': True,
+    }
+    
+    installation_mesg = """
+    To use Klusta, install klusta and klustakwik2:
+       >>> pip install klusta klustakwik
+    
+    More information on klusta at:
+      * https://github.com/kwikteam/phy"
+      * https://github.com/kwikteam/klusta
+    """
+    
+    
+    def __init__(self, recording=None, output_folder=None,
+                                    by_property=None, parallel=False):
+        BaseSorter.__init__(self, recording=recording, output_folder=output_folder,
+                                    by_property=by_property, parallel=parallel)
+
+    def set_params(self, **params):
+        self.params = params
+        
+    
+    def _setup_recording(self):
```

Comment:
I don't understand what _setup_recording does to be honest, 

- Why does it save a probe file and also save out to a binary format?

Reply:
It is split in 2 steps:
  * setup recording (basically copy/paste files), it can take long time for big files
  * and the comptation itself.

If we want to make benchmark on the second step only it will be usefull.

Reply:
So set_up recording does what exactly? I thought that the recording extractor was supposed to eliminate the need for copying over big files. Am I missing something?

Reply:
hei @colehurwitz31 
many of the spikesorters require to save the data in a specific format (e.g. dat) and or provide probe files and params files. 

The _setup_recording() is a nice way to organize these steps.

Thanks @samuelgarcia